src/Strategy/GreedyCacheStrategy.php

```diff
@@ -91,7 +93,15 @@ protected function getCacheObject(RequestInterface $request, ResponseInterface $
         }
 
         $response = $response->withoutHeader('Etag')->withoutHeader('Last-Modified');
-        return new CacheEntry($request, $response, new \DateTime(sprintf('+%d seconds', $this->ttl)));
+
+        $ttl = $this->defaultTtl;
+        if ($request->hasHeader(self::HEADER_TTL)) {
+            $ttlHeaderValues = $request->getHeader(self::HEADER_TTL);
+            $ttl = (int)reset($ttlHeaderValues);
+            unset($ttlHeaderValues);
```

Comment:
Why do you do this `unset`?

Reply:
it was just to not keep a variable that has no purposes. I would prefer to not create it but reset takes only variables. Do you prefer I remove this line @Kevinrob ?

Reply:
No, I thing that it's okay to do that. Thanks